### PR TITLE
Move parts of lib/url to calypso-url package

### PR DESCRIPTION
**Pull request by sirbrillig**
#### Changes proposed in this Pull Request

As part of my goal of moving large amounts of checkout into a package outside of calypso so that it can have type safety and be used in wp-admin, I've been working to put checkout's major dependencies into packages first. Probably the biggest of those is the combination of `lib/products-values` and `lib/plans`, but those directories have a dependency on `lib/url`, which has very few dependencies of its own. This PR therefore begins to move `lib/url` into the package `@automattic/calypso-url`. To start with, it only moves the code required by `lib/plans`, however, which is the following functions:

- `format`
- `getUrlParts`
- `getUrlFromParts`
- `determineUrlType`
- and the enum `URL_TYPE`

I'm focusing on these functions (which do seem to be the core of that directory anyway, according to its README) because moving the entirety of `lib/url` would require dealing with quite a few other dependencies, which are really out of the scope of this project (but could be migrated later).

#### Testing instructions

This does not change any of the functions that are moved, and it attempts to update all imports of those functions. If there is any import that was missed, the calypso build should fail. Therefore automated tests should be sufficient to prove that these changes are safe, but I'd appreciate any other opinions.

### CODE REVIEW

File: packages/calypso-url/src/types.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export type URL = string;
```
Comment (jsnajdr) [inline packages/calypso-url/src/types.ts]: I believe this typedef should be eventually removed from the package and that we should use `string` instead. In TypeScript with its structural typing, it doesn't make much sense to declare alternative names for types like this.

TypeScript's native typings for `URL` or `URLSearchParams` (in `lib.dom.d.ts`) also use `string` directly.

The name `URL` also conflicts with the `URL` global that's declared both in browsers and Node.

**Review by jsnajdr — APPROVED** [review]
Looks good :+1:

Just curious: why didn't you also move `addQueryArgs` or `isHttps`? Any trouble with their Lodash and type dependencies?

File: packages/calypso-url/src/types.ts (hunk 1 above)
Comment (sirbrillig) [reply packages/calypso-url/src/types.ts]: Good point. I'll update that.

**Comment by sirbrillig** [comment]
> Just curious: why didn't you also move addQueryArgs or isHttps? Any trouble with their Lodash and type dependencies?

Originally I tried moving the whole directory (#51797) but it was so complex that I decided the risk of breaking something was too great; then I started from the other direction and just moved the minimum things we need for `lib/plans`. There's no reason not to move the rest, I just thought it was safer to do it in smaller pieces.

**Pull request merged**